### ISSUE TITLE
pth2onnx error : ONNX export of operator adaptive_avg_pool1d, input size not accessible.

### ISSUE THREAD

**Issue by VivizSun** [cqylunlun/GLASS#41]
Thank you for your contributions.

When I finish training and try to convert to ONNX, an error occurs (as shown in the image below). 
What steps have I missed?

![Error](https://github.com/user-attachments/assets/fa4a3ca4-4271-4a11-8c65-2e4a5f224bce)

Environment:
absl-py                 2.1.0
cachetools              5.5.0
certifi                 2024.12.14
charset-normalizer      3.4.1
click                   8.1.7
colorama                0.4.6
coloredlogs             15.0.1
contourpy               1.3.1
cuda-python             11.8.2
cycler                  0.12.1
Cython                  3.0.11
filelock                3.16.1
flatbuffers             24.12.23
fonttools               4.55.3
fsspec                  2024.12.0
google-auth             2.37.0
google-auth-oauthlib    1.2.1
grpcio                  1.68.1
huggingface-hub         0.27.0
humanfriendly           10.0
idna                    3.10
imageio                 2.36.1
Jinja2                  3.1.5
joblib                  1.4.2
kiwisolver              1.4.8
lazy_loader             0.4
Markdown                3.7
markdown-it-py          3.0.0
MarkupSafe              3.0.2
matplotlib              3.8.2
mdurl                   0.1.2
mpmath                  1.3.0
networkx                3.4.2
numpy                   1.26.3
oauthlib                3.2.2
onnx                    1.16.2
onnxruntime-gpu         1.18.1
onnxsim                 0.4.36
opencv-python-headless  4.10.0.84
packaging               24.2
pandas                  1.5.2
pillow                  10.2.0
pip                     24.3.1
protobuf                4.23.4
pyasn1                  0.6.1
pyasn1_modules          0.4.1
Pygments                2.18.0
pyparsing               3.2.0
pyreadline3             3.5.4
python-dateutil         2.9.0.post0
pytz                    2024.2
PyYAML                  6.0.2
requests                2.32.3
requests-oauthlib       2.0.0
rich                    13.9.4
rsa                     4.9
safetensors             0.4.5
scikit-image            0.22.0
scikit-learn            1.4.0
scipy                   1.11.4
setuptools              58.1.0
six                     1.17.0
sympy                   1.13.1
tensorboard             2.15.1
tensorboard-data-server 0.7.2
threadpoolctl           3.5.0
tifffile                2024.12.12
timm                    0.9.12
torch                   2.5.1+cu124
torchaudio              2.5.1+cu124
torchvision             0.20.1+cu124
tqdm                    4.66.1
typing_extensions       4.12.2
urllib3                 2.3.0
Werkzeug                3.1.3

**Comment by cqylunlun** [comment]
Thank you for the recognition. It seems that you are using an earlier version of our code. Please check the modifications made in the latest version of `glass.py` at [L165](https://github.com/cqylunlun/GLASS/blob/85c4e2f8529c6548144271c151d6bdd822547fbd/glass.py#L165) and [L178](https://github.com/cqylunlun/GLASS/blob/85c4e2f8529c6548144271c151d6bdd822547fbd/glass.py#L178).

**Comment by VivizSun** [comment]
Thank you very much.
The conversion has been successful. ^^
